### PR TITLE
Use ActionLauncher for positron annihilation, Rayleigh scattering, and Seltzer-Berger models

### PR DESCRIPTION
**Pull request by stognini**
This PR is a follow-up of #783 and refactors `EPlusGGModel`, `RayleighModel`, and `SeltzerBergerModel`.

### CODE REVIEW

File: src/celeritas/em/model/EPlusGGModel.cu [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * Interact with device data.
+ */
+void EPlusGGModel::execute(CoreParams const& params,
+                           CoreStateDevice& state) const
+{
+    auto execute = make_action_track_executor(
+        params.ptr<MemSpace::native>(),
+        state.ptr(),
+        this->action_id(),
+        InteractionApplier{EPlusGGExecutor{this->host_ref()}});
```
Comment (sethrj) [inline src/celeritas/em/model/EPlusGGModel.cu]: Should be `this->device_ref()`. The use of `host_ref` in the `.cu` file I wrote was a mistake, as we discussed with Philippe. It returns the same data type for now so probably shouldn't matter?

File: src/celeritas/em/model/RayleighModel.cu [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Interact with device data.
+ */
+void RayleighModel::execute(CoreParams const& params,
+                            CoreStateDevice& state) const
+{
+    auto execute = make_action_track_executor(
+        params.ptr<MemSpace::native>(),
+        state.ptr(),
+        this->action_id(),
+        InteractionApplier{RayleighExecutor{this->host_ref()}});
```
Comment (sethrj) [inline src/celeritas/em/model/RayleighModel.cu]: Same comment about device data

File: src/celeritas/em/model/SeltzerBergerModel.cu [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * Interact with device data.
+ */
+void SeltzerBergerModel::execute(CoreParams const& params,
+                                 CoreStateDevice& state) const
+{
+    auto execute = make_action_track_executor(
+        params.ptr<MemSpace::native>(),
+        state.ptr(),
+        this->action_id(),
+        InteractionApplier{SeltzerBergerExecutor{this->host_ref()}});
```
Comment (sethrj) [inline src/celeritas/em/model/SeltzerBergerModel.cu]: Same

**Review by sethrj — CHANGES_REQUESTED** [review]

**Review by sethrj — APPROVED** [review]
LGTM! I'll wait on a successful pipeline. Thanks!

**Comment by sethrj** [comment]
@stognini can you make sure this compiles on wildstyle before your next push?

**Comment by stognini** [comment]
> @stognini can you make sure this compiles on wildstyle before your next push?

The last commit was actually from wildstyle and it passed all tests...

**Comment by sethrj** [comment]
Sorry @stognini , that was totally on me... I messed up the merge of develop into the branch 🤦

**Pull request merged**